chain/builder.go

```diff
@@ -444,7 +447,12 @@ func BuildBlock(
 	timestampKey := TimestampKey(b.vm.StateManager().TimestampKey())
 	timestampKeyStr := string(timestampKey)
 	feeKeyStr := string(feeKey)
-	tsv := ts.NewView(set.Of(heightKeyStr, timestampKeyStr, feeKeyStr), map[string][]byte{
+
+	heightKeyKey := state.NewKey(heightKeyStr, state.Read, state.Write)
```

Comment:
TODO make sure these scopes are all correct

Reply:
When I had this convo with Patrick, my understanding is that it would be read at the beginning of the block and then written later